tests/operators/test_convection.py

```diff
@@ -3,22 +3,33 @@
 """Tests for convection diagnostics."""
 
 import iris
+import numpy as np
 
 import CSET.operators.convection as convection
 
 
 def test_cape_ratio():
     """Compare with precalculated ratio."""
-    precalculated = iris.load_cube("tests/test_data/convection/ECFlagB.nc")
-    precalculated_2 = iris.load_cube("tests/test_data/convection/ECFlagB_2.nc")
     SBCAPE = iris.load_cube("tests/test_data/convection/SBCAPE.nc")
     MUCAPE = iris.load_cube("tests/test_data/convection/MUCAPE.nc")
     MUCIN = iris.load_cube("tests/test_data/convection/MUCIN.nc")
-    assert (
-        convection.cape_ratio(SBCAPE, MUCAPE, MUCIN).data.all()
-        == precalculated.data.all()
-    )
-    assert (
-        convection.cape_ratio(SBCAPE, MUCAPE, MUCIN, MUCIN_thresh=-1.5).data.all()
-        == precalculated_2.data.all()
+    cape_75 = convection.cape_ratio(SBCAPE, MUCAPE, MUCIN)
+    precalculated_75 = iris.load_cube("tests/test_data/convection/ECFlagB.nc")
+    assert np.allclose(cape_75.data, precalculated_75.data, atol=1e-5, equal_nan=True)
+    # TODO: Test data clobbered by -75, so disabled until regenerated.
+    # It seems to be passing for some reason...? Maybe too tolerant?
+    cape_1p5 = convection.cape_ratio(SBCAPE, MUCAPE, MUCIN, MUCIN_thresh=-1.5)
+    precalculated_1p5 = iris.load_cube("tests/test_data/convection/ECFlagB_2.nc")
+    assert np.allclose(cape_1p5.data, precalculated_1p5.data, atol=1e-5, equal_nan=True)
```

Comment:
Something mysterious is going on here. The test data for a threshold of -1.5 in `ECFlagB_2.nc` is exactly the same as the data for -75.0 in `ECFlag_B.nc`. The files are bit-for-bit identical. And yet this test now passes

I have got fairly large tolerances (within 0.0001), but is that all a change between a threshold of -1.5 and -75.0 would amount to?

@daflack it would be good to investigate if you expect these two thresholds to give the same results.

Reply:
Further investigation reveals that they are exactly equal, even with a tolerance of 0.